job_definitions/docker_base_images.yml

```diff
@@ -14,24 +27,39 @@
       node {
         try {
           stage('Prepare') {
-            git url: "git@github.com:alphagov/digitalmarketplace-docker-base.git", branch: 'master', credentialsId: 'github_com_and_enterprise', poll: true
+            currentBuild.displayName = "#${BUILD_NUMBER} - ${HEAD_BRANCH}"
+            git url: "git@github.com:alphagov/digitalmarketplace-docker-base.git", branch: HEAD_BRANCH, credentialsId: 'github_com_and_enterprise', poll: true
           }
           stage('Build') {
             sh("make build")
           }
           stage('Push') {
             docker_credentials = sh(script: '/var/lib/jenkins/digitalmarketplace-credentials/sops-wrapper -d /var/lib/jenkins/digitalmarketplace-credentials/jenkins-vars/docker_credentials_env.enc', returnStdout: true).trim()
             withEnv(docker_credentials.tokenize("\n")) {
-                sh("set +x; docker login -u ${DOCKER_USERNAME} -p ${DOCKER_PASSWORD}")
+              sh("set +x; docker login -u ${DOCKER_USERNAME} -p ${DOCKER_PASSWORD}")
+            }
+
+            if (HEAD_BRANCH != "master" && sh(script: "cat VERSION", returnStdout: true) ==~ /\d.*/) {
+              input("WARNING: VERSION looks numeric - it is probably a bad idea to create a release- tagged image from a non-master HEAD_BRANCH!")
```

Comment:
When would the `VERSION` file not be numeric?

Reply:
When I'm tinkering around and making it `ris-experiment-12` for instance.

Reply:
If you don't like us having non-numeric tags, I wouldn't look at our base images on dockerhub right now...

Reply:
It was just the least invasive way I could see to do it.

Reply:
Maybe I should take another look at alphagov/digitalmarketplace-docker-base#52 :P